### DISCUSSION TITLE
filtering rules

### DISCUSSION THREAD

**Discussion by ale82x** [healthchecks/healthchecks#1081] [Q&A] [answered]
Hello, i need to track some email.
but i have this auto generated email (i can't change template)

i want setup 
 Success Keywords : [0] failures
but in
 Failure Keywords : NOT [0] failures
because failures cloud be 
[1] failures
[2] failures
...
[999] failures

so i can use NOT [0] or something else?

thank you

**Comment by cuu508** [comment]
You can try the following:

Success keywords: `0`
Failure keywords: `1,2,3,4,5,6,7,8,9`

The string `[0] failures` will match success keywords, as it contains "0".

Strings like `[999] failures` will match failure keywords, as they contain non-zero digits.

**Comment by ale82x** [comment, marked as answer]
big thank you

i try to 
Success keywords: [0] failures
Failure keywords: 1,2,3,4,5,6,7,8,9

because i set only 0 i think failures is 10 becomes success.

it's ok?

thank you

**Reply by cuu508** [reply]
Ah, good thinking, but I think this will still be alright even if you set bare "0" as the success keyword.

When Healthchecks checks for the filtering keywords, it first looks for the presence of failure keywords. If none found, it then looks for success keywords, and then for start keywords.

So in the example of "[10] failures" it will first match "1" as a failure keyword. It will declare it a failure and will not check further keywords. 

You can test the different scenarios out by creating a dummy check, configuring its filtering rules, and then sending email messages to it from your normal email client.

**Comment by ale82x** [comment]
ah , good to know.
I'll write it down for future!!!! :)
big thank you
